inference-engine/tests/functional/plugin/cpu/single_layer_tests/activation.cpp

```diff
@@ -0,0 +1,155 @@
+            strExpectedPrc = "FP32";
+        }
+
+        std::string isaType;
+        if (with_cpu_x86_avx512f()) {
+            isaType = "jit_avx512";
+        } else if (with_cpu_x86_avx2()) {
+            isaType = "jit_avx2";
+        } else if (with_cpu_x86_sse42()) {
+            isaType = "jit_sse42";
+        } else {
+            isaType = "ref";
+        }
+        selectedType = isaType + "_" + strExpectedPrc;
+
+        // Withing the test scope we don't need any implicit bf16 optimisations, so let's run the network as is.
+        configuration.insert({PluginConfigParams::KEY_ENFORCE_BF16, PluginConfigParams::NO});
+
+        activationType = activationDecl.first;
+        auto constantsValue = activationDecl.second;
+        auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+        auto params = ngraph::builder::makeParams(ngPrc, {shapes.first});
+        auto activation = ngraph::builder::makeActivation(params[0], ngPrc, activationType, shapes.second, constantsValue);
+        activation->get_rt_info() = getCPUInfo();
+        function = std::make_shared<ngraph::Function>(ngraph::NodeVector{activation}, params, "Activation");
+    }
+};
+
+TEST_P(ActivationLayerCPUTest, CompareWithRefs) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+
+    Run();
+    CheckCPUImpl(executableNetwork, "Eltwise");
+}
+
+
+namespace {
+// list only types supported by eltwise
+const std::map<ActivationTypes, std::vector<std::vector<float>>> activationTypes = {
+        {Sqrt,        {{}}},
+        {Sigmoid,     {{}}},
+        {Tanh,        {{}}},
+        {Relu,        {{}}},
+        {Gelu,        {{}}},
+        {Exp,         {{}}},
+        {Clamp,       {{-2.0f, 2.0f}}},
+        {Elu,         {{0.1f}}},
+        {Swish,       {{0.1f}}},
+        {HSwish,      {{}}},
+        {Mish,        {{}}}
+};
+
+const std::map<ActivationTypes, std::vector<std::vector<float>>> activationParamTypes = {
+        {PReLu, {{-0.01f}}},
+};
+
+std::map<std::vector<size_t>, std::vector<std::vector<size_t>>> basic = {
+        {{1, 50}, {{}}},
+        {{1, 128}, {{}}},
+};
+
+std::map<std::vector<size_t>, std::vector<std::vector<size_t>>> preluBasic = {
+        {{1, 50}, {{1}, {50}}},
+        {{1, 128}, {{1}, {128}}},
+};
+
+std::vector<Precision> bf16InpOutPrc = {Precision::BF16, Precision::FP32};
+
+const auto basicCases = ::testing::Combine(
+        ::testing::Combine(
+            ::testing::ValuesIn(CommonTestUtils::combineParams(activationTypes)),
+            ::testing::Values(Precision::BF16),
+            ::testing::ValuesIn(bf16InpOutPrc),
+            ::testing::ValuesIn(bf16InpOutPrc),
+            ::testing::Values(InferenceEngine::Layout::ANY),
+            ::testing::Values(InferenceEngine::Layout::ANY),
+            ::testing::ValuesIn(CommonTestUtils::combineParams(basic)),
+            ::testing::Values(CommonTestUtils::DEVICE_CPU)),
+        ::testing::Values(emptyCPUSpec)
```

Comment:
Don't we want to test different layouts here?

Reply:
Done in c626e2ef686f333b4836992d26e9ef86b60ba789.